### PR TITLE
v1.16 Backports 2024-08-19

### PR DESCRIPTION
**Pull request by viktor-kurchenko**
* [x] #34076 (@chaunceyjiang)
 * [ ] #34234 (@tamilmani1989)
 * [x] #34285 (@julianwiedmann)
 * [x] #34270 (@rastislavs)
 * [x] #34227 (@dylandreimerink)
 * [x] #34303 (@julianwiedmann)
 * [x] #34336 (@aanm)
 * [x] #34339 (@sayboras)
 * [x] #34236 (@mhofstetter) :warning: resolved conflicts
 * [x] #34282 (@chancez) :warning: resolved conflicts
 * [x] #34229 (@chancez)
 * [x] #34335 (@rastislavs)
 * [x] #34346 (@jrajahalme)
 * [ ] #34372 (@Artyop)
 * [x] #33625 (@julianwiedmann) :warning: resolved conflicts
 * [x] #34415 (@harsimran-pabla)
 * [x] #34420 (@julianwiedmann) :warning: resolved conflicts
 * [x] #34400 (@mhofstetter)
 * [x] #34361 (@chancez)
 * [x] #34388 (@pchaigno)

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 34076 34231 34234 34285 34270 34227 34303 34336 34339 34236 34282 34229 34335 34346 34372 33625 34415 34420 34400 34361 34388
```

### CODE REVIEW

**Review by sayboras — APPROVED** [review]
Thanks and looks good for my commit.

**Comment by viktor-kurchenko** [comment]
/test-backport-1.16

**Review by mhofstetter — APPROVED** [review]
My commits look good - thanks for fixing the conflicts @viktor-kurchenko :pray:

**Review by harsimran-pabla — APPROVED** [review]

**Review by pchaigno — APPROVED** [review]

**Review by chancez — APPROVED** [review]

**Review by glibsm — APPROVED** [review]

**Comment by harsimran-pabla** [comment]
Reviewed @rastislavs  PRs since he is on 🏖️

**Review by dylandreimerink — APPROVED** [review]

**Review by ldelossa — APPROVED** [review]

File: .github/workflows/tests-datapath-verifier.yaml [hunk 1]
```diff
@@ -100,6 +100,9 @@ jobs:
             ci-kernel: '61'
           # renovate: datasource=docker depName=quay.io/lvh-images/complexity-test
           - kernel: '6.6-20240710.064909'
+            ci-kernel: '61'
+          # renovate: datasource=docker depName=quay.io/lvh-images/complexity-test
+          - kernel: 'bpf-next-20240711.013133'
```
Comment (julianwiedmann) [inline .github/workflows/tests-datapath-verifier.yaml]: To preserve the current branch content, this should be:

```suggestion
          - kernel: 'kernel: '6.6-20240710.064909''
```

(in stable branches, we replaced the bpf-next kernel with a stable kernel)

**Review by julianwiedmann — CHANGES_REQUESTED** [review]
Just one tiny fix-up needed in the conflict resolution.

File: .github/workflows/tests-datapath-verifier.yaml (hunk 1 above)
Comment (viktor-kurchenko) [reply .github/workflows/tests-datapath-verifier.yaml]: @julianwiedmann cool catch! thank you!!!
Shouldn't I remove the following lines at all:
```
          # renovate: datasource=docker depName=quay.io/lvh-images/complexity-test
          - kernel: 'bpf-next-20240711.013133'
            ci-kernel: 'netnext'
```
?

If I replace `'bpf-next-20240711.013133'` with `'6.6-20240710.064909'` it looks like the previous kernel version...

Comment (julianwiedmann) [reply .github/workflows/tests-datapath-verifier.yaml]: I'd expect that when backporting https://github.com/cilium/cilium/pull/34420 and inserting the diff in the right spot, we end up with the following:

``` 
          # renovate: datasource=docker depName=quay.io/lvh-images/complexity-test
          - kernel: '6.1-20240710.064909'
            ci-kernel: '61'
          # renovate: datasource=docker depName=quay.io/lvh-images/complexity-test
          - kernel: '6.6-20240710.064909'
            ci-kernel: '61'
          # renovate: datasource=docker depName=quay.io/lvh-images/complexity-test
          - kernel: '6.6-20240710.064909'
            ci-kernel: 'netnext'
```

Which is three distinct test configs (the `ci-kernel` parameter selects what datapath configs should be tested on-top of the specified kernel)

Comment (viktor-kurchenko) [reply .github/workflows/tests-datapath-verifier.yaml]: Gotcha, fixed! Thank you!

**Comment by viktor-kurchenko** [comment]
/test-backport-1.16

File: .github/workflows/tests-datapath-verifier.yaml (hunk 1 above)
Comment (julianwiedmann) [reply .github/workflows/tests-datapath-verifier.yaml]: looks good now, thank you!

**Review by julianwiedmann — APPROVED** [review]

**Comment by viktor-kurchenko** [comment]
@sayboras may I ask you to review https://github.com/cilium/cilium/pull/34336 changes instead of André (he is on vacation).

**Review by sayboras — APPROVED** [review]
Looks good for #34336 as well

**Comment by viktor-kurchenko** [comment]
@learnitall could you please review https://github.com/cilium/cilium/pull/34234 changes?

**Comment by viktor-kurchenko** [comment]
@squeed could you please review https://github.com/cilium/cilium/pull/34346 changes?

**Review by jrajahalme — APPROVED** [review]
Looking good for my PR

**Review by aanm — CHANGES_REQUESTED** [review]
I would suggest to drop the commit https://github.com/cilium/cilium/pull/34452/commits/58ccc89992e31eef367c0b972c23dbaf26951723. The changes in this PR are a lot different from the upstream PR so it would make sense to let @hemanthmalla  to perform the backport himself.

**Comment by viktor-kurchenko** [comment]
> I would suggest to drop the commit [58ccc89](https://github.com/cilium/cilium/commit/58ccc89992e31eef367c0b972c23dbaf26951723). The changes in this PR are a lot different from the upstream PR so it would make sense to let @hemanthmalla to perform the backport himself.

Done, thank you!

**Comment by viktor-kurchenko** [comment]
/test-backport-1.16

**Comment by hemanthmalla** [comment]
Yeah, some of the changes didn't really look relevant to my original commit and I didn't need to do those for our 1.14 / 1.15 backports. I'll handle this one separately. Thanks for looking into this @viktor-kurchenko / @aanm

**Review by aanm — APPROVED** [review]

**Review by Artyop — APPROVED** [review]

**Review by joestringer — APPROVED** [review]

**Pull request merged**